Sources/TuistCache/ContentHashing/CopyFilesContentHasher.swift

```diff
@@ -0,0 +1,29 @@
+import Foundation
+import TuistCore
+
+public protocol CopyFilesContentHashing {
+    func hash(copyFiles: [CopyFilesAction]) throws -> String
+}
+
+/// `CopyFilesContentHasher`
+/// is responsible for computing a unique hash that identifies a list of CopyFilesAction models
+public final class CopyFilesContentHasher: CopyFilesContentHashing {
+    private let contentHasher: ContentHashing
+
+    // MARK: - Init
+
+    public init(contentHasher: ContentHashing) {
+        self.contentHasher = contentHasher
+    }
+
+    // MARK: - CopyFilesContentHashing
+
+    public func hash(copyFiles: [CopyFilesAction]) throws -> String {
+        var stringsToHash: [String] = []
+        for action in copyFiles {
+            let fileHashes = try action.files.map { try contentHasher.hash(path: $0.path) }
```

Comment:
Not sure if it's possible in the case of copy files, but with scripts, we couldn't hash the files because paths can contain variables, e.g. ${SRCROOT}, that are resolved at build time.

Reply:
I think for files this should be ok, like the list of resources.